pypdf/_xobj_image_helpers.py

```diff
@@ -195,7 +195,13 @@ def bits2byte(data: bytes, size: Tuple[int, int], bits: int) -> bytes:
         else:
             if img.mode == "1":
                 # Two values ("high" and "low").
-                assert len(lookup) == 2 * nb, len(lookup)
+                expected_count = 2 * nb
+                if len(lookup) != expected_count:
+                    if len(lookup) < expected_count:
+                        raise PdfReadError(f"Not enough lookup values: Expected {expected_count}, got {len(lookup)}.")
+                    lookup = lookup[:expected_count]
+                    if not all(_value in WHITESPACES for _value in lookup[expected_count:]):
+                        raise PdfReadError(f"Too many lookup values: Expected {expected_count}, got {len(lookup)}.")
```

Comment:
```suggestion
                    if not all(_value in WHITESPACES for _value in lookup[expected_count:]):
                        raise PdfReadError(f"Too many lookup values: Expected {expected_count}, got {len(lookup)}.")
                    lookup = lookup[:expected_count]
```

Shouldn't it be in this order ?

Reply:
This has not yet been merged and is part of another/my PR, so I see no real value of including and discussing it here in your apparently completely unrelated PR.